Project/api/src/tests/unitTests/db/order.test/retrieveOrders.test.ts

```
@@ -0,0 +1,191 @@
+            totalPrice,
+            timestamp
+        );
+
+        dummyOrder2 = {
+            ...(dummyOrder2 as Order),
+            status: 2,
+            employeeID: new ObjectId('672df427f54107237ff75569'),
+        };
+
+        await orderRepository.save(dummyOrder1);
+        await orderRepository.save(dummyOrder2);
+    });
+
+    afterEach(async () => {
+        const repository = AppDataSource.getRepository(Order);
+        await repository.delete({}); //Deletes all documents in the collection
+    });
+
+    afterAll(async () => {
+        await AppDataSource.destroy();
+    });
+
+    it('should get all accepted orders', async () => {
+        if (!dummyOrder1 || !dummyOrder2) throw new Error('An order was not created properly!');
+
+        const orders = await orderAndFeedbackService.getAllAcceptedOrders();
+
+        expect(orders).not.toBeNull();
+
+        expect(orders).toEqual(
+            expect.arrayContaining([
+                expect.objectContaining({
+                    _id: expect.any(Object), // If the ID is autogenerated
+                    customerID: dummyOrder1.customerID,
+                    restaurantID: dummyOrder1.restaurantID,
+                    address: dummyOrder1.address,
+                    totalPrice: dummyOrder1.totalPrice,
+                    orderItemList: dummyOrder1.orderItemList,
+                    timestamp: dummyOrder1.timestamp,
+                }),
+                expect.objectContaining({
+                    _id: expect.any(Object),
+                    customerID: dummyOrder2.customerID,
+                    restaurantID: dummyOrder2.restaurantID,
+                    address: dummyOrder2.address,
+                    totalPrice: dummyOrder2.totalPrice,
+                    orderItemList: dummyOrder2.orderItemList,
+                    timestamp: dummyOrder2.timestamp,
+                }),
+            ])
+        );
+    });
+
+    it('should handle errors and return null', async () => {
```

If the database returns an error, we shouldn't return null, we should return the error and let the caller handle it.